rescompy/rescompy.py

```diff
@@ -1021,14 +1090,14 @@ def train(
             if len(inputs) == 1: msg1 = 'inputs'
             else: msg1 = f'inputs[{task_ind}]'
             utils.check_shape(inputs[task_ind].shape,
-                          (None, self.input_dimension), msg1)            
-
-        num_samples = len(inputs)
-        batch_size = min(num_samples, batch_size)
+                          (None, self.input_dimension), msg1)
+		
+        # If the feature_function has a compiled option, use it. Otherwise
+		# use the non-compiled version.
+        if hasattr(feature_function, 'compiled'):
```

Comment:
Would this be better as:

if hasattr(feature_function, 'compiled') and \
			issubclass(type(feature_function), features.SingleFeature):
    feature_function_jit = feature_function.compiled
else: feature_function_jit = feature_function

Jit-compilation of the feature function gave me about 10% quicker run-times in a few MRS-style tasks that I tried.  For tasks with fewer sample input signals compilation slowed training slightly.  My guess is that most classification-style tasks will require enough sample inputs for training to make the compilation worth it, but most time-series prediction tasks will use fewer sample inputs. So maybe the default should be to jit-compile only when using a features.SingleFeature feature function.